### PR TITLE
[RFC] doc/install: Rethink doc/install

### PR DESCRIPTION
**Pull request by sebastian-philipp**
What about having a list of the various installation possibilities?

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/install/index.rst [hunk 1]
```diff
@@ -1,72 +1,74 @@
...
-=======================
+.. _install-overview:
 
+===============
+Installing Ceph
+===============
 
-Get Software
+There are different options to install Ceph. Each solution meet different
+requirements, like ease of maintenance.
```
Comment (liewegas) [inline doc/install/index.rst]: Each method tends to meet different requirements, like ease of maintenance, and many solutions are better suited for different versions.  Please see the notes associated with each method before choosing.

File: doc/install/index.rst [hunk 2]
```diff
@@ -1,72 +1,74 @@
...
 
-There are several methods for getting Ceph software. The easiest and most common
-method is to `get packages`_ by adding repositories for use with package
-management tools such as the Advanced Package Tool (APT) or Yellowdog Updater,
-Modified (YUM). You may also retrieve pre-compiled packages from the Ceph
-repository. Finally, you can retrieve tarballs or clone the Ceph source code
-repository and build Ceph yourself.
+Installs Ceph using Ansible.
 
+* `docs.ceph.com/ceph-ansible <https://docs.ceph.com/ceph-ansible/>`_
```
Comment (liewegas) [inline doc/install/index.rst]: Caveats:
- Ansible is not integrated with the orchestration API (new in octopus), so new management features in the CLI and dashboard will not work.

File: doc/install/index.rst [hunk 3]
```diff
@@ -1,72 +1,74 @@
...
 
-	Get Packages <get-packages>
-	Get Tarballs <get-tarballs>
-	Clone Source <clone-source>
-	Build Ceph <build-ceph>
-    	Ceph Mirrors <mirrors>
-	Ceph Containers <containers>
+Install ceph using ceph-deploy
 
+* :ref:`ceph-deploy-index`
```
Comment (liewegas) [inline doc/install/index.rst]: Caveats:
- ceph-deploy is no longer actively maintained.
- ceph-deploy is not tested on versions newer than Nautilus.
- ceph-deploy does not support RHEL/CentOS 8 or newer.

File: doc/install/index.rst [hunk 4]
```diff
@@ -1,72 +1,74 @@
...
 
-.. toctree::
-   :maxdepth: 1
+Installs Ceph using containers and systemd.
 
-	Install ceph-deploy <install-ceph-deploy>
-    	Install Ceph Storage Cluster <install-storage-cluster>
-	Install Ceph Object Gateway <install-ceph-gateway>
-	Install Virtualization for Block <install-vm-cloud>
+* :ref:`cephadm-bootstrap`
```
Comment (liewegas) [inline doc/install/index.rst]: Caveats:
- cephadm is only supported on Octopus (and newer) releases.
- cephadm is fully integrated with the new orchestration API, and fully supports the new CLI and dashboard features to manage the cluster deployment.
- cephadm requires container support (podman or docker) and Python 3 (available on almost all modern Linux distros).

File: doc/install/index.rst [hunk 5]
```diff
@@ -1,72 +1,74 @@
...
-Install Software
-================
 
-Once you have the Ceph software (or added repositories), installing the software
-is easy. To install packages on each :term:`Ceph Node` in your cluster. You may
-use  ``ceph-deploy`` to install Ceph for your storage cluster, or use package
-management tools. You should install Yum Priorities for RHEL/CentOS and other
-distributions that use Yum if you intend to install the Ceph Object Gateway or
-QEMU.
+cephadm
```
Comment (liewegas) [inline doc/install/index.rst]: Let's move this to the top?

File: doc/install/index.rst [hunk 6]
```diff
@@ -1,72 +1,74 @@
...
+Puppet
+======
 
-.. _get packages: ../install/get-packages
+Installs Ceph using Puppet
+
+* `github.com/openstack/puppet-ceph <https://github.com/openstack/puppet-ceph>`_
+
+
+Rook
```
Comment (liewegas) [inline doc/install/index.rst]: And this one second?

File: doc/install/index.rst [hunk 7]
```diff
@@ -1,72 +1,74 @@
...
+====
+
+Installs Ceph in Kubernetes.
+
+* `rook.io <https://rook.io/>`_ 
+
```
Comment (liewegas) [inline doc/install/index.rst]: Caveats:
- Rook only supports Nautilus and newer releases.
- Rook is the preferred deployment method for Ceph with Kubernetes.
- Rook fully supports the new orchestrator API, and new management features in the CLI and dashboard are fully supported.

File: doc/install/index.rst (hunk 2 above)
Comment (sebastian-philipp) [reply doc/install/index.rst]: Well, we need good way to hint users to the right tool. E.g. k8s has a table of features: https://kubernetes.io/docs/setup/#production-environment

File: doc/install/index.rst (hunk 5 above)
Comment (sebastian-philipp) [reply doc/install/index.rst]: Then, how do you want to order the list of tools? Right now, this is alphabetical.

File: doc/install/index.rst (hunk 1 above)
Comment (zdover23) [reply doc/install/index.rst]: Each method meets different requirements (for instance, some result in Ceph clusters that are easier to maintain). Review the documentation for each method before choosing one.

File: doc/install/index.rst (hunk 2 above)
Comment (zdover23) [reply doc/install/index.rst]: We need a list of all caveats of this kind.

File: doc/install/index.rst (hunk 7 above)
Comment (zdover23) [reply doc/install/index.rst]: s/Rook only supports/Rook supports only
s/the new orchestrator API, and new management features in the CLI and dashboard are fully supported/the new orchestrator API, the new management features in the CLI and dashboard/

**Comment by sebastian-philipp** [comment]
ping @zdover23 ?

File: doc/install/index.rst (hunk 1 above)
Comment (sebastian-philipp) [reply doc/install/index.rst]: do you have an idea of how to accomplish this? A table? Just extend each sections with a list of caveats + requirements?

File: doc/install/index.rst (hunk 5 above)
Comment (liewegas) [reply doc/install/index.rst]: maybe break into two sections... Recommended (with cephadm and rook) and then Other (with everything else, alphabetical?)

File: doc/install/index.rst (hunk 2 above)
Comment (zdover23) [reply doc/install/index.rst]: @sebastian-philipp This PR is now superseded by this one:https://github.com/ceph/ceph/pull/33890

If you're satisfied with this, we can close this PR.

**Pull request closed without merging**